tensorflow/compiler/tf2tensorrt/utils/trt_shape_optimization_profiles.h

```diff
@@ -143,30 +164,55 @@ class TrtShapeOptimizationProfile {
       nvinfer1::ICudaEngine* engine,
       std::vector<TrtUniquePtrType<nvinfer1::IExecutionContext>>& exec_context);
 
-  // Maps input vector shapes to TRT Optimization profiles (min, max, opt) i.e.
-  // maps input_shapes_ to profiles_
-  void InitProfiles();
+  // Creates optimization profiles profiles_ for the set of concrete input
+  // shapes collected in input_shapes_. The input_partial_shapes of the network
+  // is used to ensure that the created optimization profiles are compatible
+  // with the network.
+  void InitProfiles(
+      const std::vector<PartialTensorShape>& input_partial_shapes);
 
   // Returns number of created profiles.
   int GetNumProfiles() const;
 
-  // Restores profiles from the engine (used after deserialization)
+  // Restores profiles from the engine (used after deserialization).
   Status RestoreProfiles(const nvinfer1::ICudaEngine* engine);
 
+  // Whether the network has any shape tensors.
+  bool HasShapeTensor() { return has_shape_tensor_; }
```

Comment:
I don't see any use of this routine, I might missed it. If we do need this info, shall we remove the field instead use absl::c_any_of(is_shape_tensor_, [](bool b) { return b; })?